### PR TITLE
Fix renaming and saving of HtmlTableSheets from HtmlTablesSheets

### LINKED ISSUE #1339: Impossible to set current column for selected rows to incremental values and to bulk save all sheets
**Small description**

Hi, I want to open and bulk save all the html tables I have [here](https://lab24.ilsole24ore.com/ecosistema-urbano/indexT.php). All the tables have the same name, then I need to rename them using `g+i`, and then save them in bulk `g+CTRL+s`.

When I run `g+i` I have `NameError: name 'setitem' is not defined`.

If I change the name of the tables, in the saved HTML files, when I run `g+CTRL+s`, I have `AttributeError: 'str' object has no attribute 'name'`

I start from 

```
 vd https://lab24.ilsole24ore.com/ecosistema-urbano/indexT.php --save-filetype=csv
```

**Expected result**

- to have the tables renamed;
- to be able to save the tables in multiple CSV files:

**Actual result with screenshot**
image

image

image

image

**Steps to reproduce with sample data and a .vd**
Please attach the commandlog (saved with `Ctrl-D`) to show the steps that led to the issue.
See [here](http://visidata.org/docs/save-restore/) for more details.

```
sheet   col     row     longname        input   keystrokes      comment
global          null_value      set-option
global          save_filetype   set-option      csv
                        open-file       https://lab24.ilsole24ore.com/ecosistema-urbano/indexT.php      o
indexT                  select-rows             gs      select all rows
indexT  name            setcol-incr             gi      set current column for selected rows to incremental values
```

**Additional context**
Please include the version of VisiData: the latest dev saul.pw/VisiData v2.9dev

### PR DESCRIPTION
**Pull request by anjakefala**
Closes #1339

### CODE REVIEW

File: visidata/basesheet.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def options(self):
 
     def __init__(self, *names, **kwargs):
         self._name = None   # initial cache value necessary for self.options
-        self.names = names
+        self.names = list(names)
```
Comment (frosencrantz) [inline visidata/basesheet.py]: I was looking at this report and found the same bugs.  I do not know the code very well, and was wondering does the getter/setter from visidata/sheets line 949  work?  Will changing the name from the Index sheet change the name of the file that is saved?  It updates names[] but it doesn't update name.  I don't know how the two fields names and name are supposed to interact. It seems like one is calculated from the other looking at this line and the next.

https://github.com/saulpw/visidata/blob/fb0a6ad18247586b9c0a6b38e708abfebd14dc44/visidata/sheets.py#L949

Comment (anjakefala) [reply visidata/basesheet.py]: You are absolutely right! It changes the sheet name through the `SheetsSheet`, but not for html tables, at least.

Comment (anjakefala) [reply visidata/basesheet.py]: Thank you! I think I got a much better fix now.

**Review by saulpw — APPROVED** [review]

File: visidata/basesheet.py (hunk 1 above)
Comment (frosencrantz) [reply visidata/basesheet.py]: Glad I could help!  Yes, I like this fix better, and I learned something.

**Pull request merged**